src/rhsm/connection.py

```diff
@@ -786,6 +819,9 @@ def _request(self, request_type, method, info=None, headers=None, cert_key_pairs
         else:
             body = None
 
+        if self.__conn is not None:
+            self.headers["Connection"] = "keep-alive"
```

Comment:
Based on my understanding of the documentation, keep-alive is used by default in HTTP 1.1 (without actually having to set the header), and only HTTP 1.0 needs to use it (or otherwise the connection is closed by the server). Does subman operate with HTTP 1.0? If only HTTP 1.1 is used you could probably just re-use the connection without setting this header on the request (unless the server actually sends back `Connection: close`.

Reply:
I can confirm that we use HTTP 1.1. Thus it is not necessary to use this HTTP header.

On the other side, when we use Python package `requests` for communication with cloud metadata servers, then this package automatically add HTTP header `Connection: keep-alive` to requests despite HTTP 1.1 is used. We would like to replace `http.client` with `requests` in the future.

Reply:
It might be because they want to cover both http versions (1.0 and 1.1) without the need for special conditions. It doesn't hurt to add the header so I wouldn't keep this PR from merge for it